giskard/utils/worker_pool.py

```diff
@@ -259,31 +266,32 @@ def spawn_worker(self):
         self.processes[p.pid] = p
         LOGGER.info("Starting a new worker %s", p.pid)
 
-    def kill_task(self, task_id: str) -> Optional[BaseException]:
+    def kill_task(self, job_id: UUID, reason: KillReason) -> Optional[BaseException]:
         # Task has timed out, we should kill it
         try:
-            future = self.futures_mapping.pop(task_id, None)
+            future = self.futures_mapping.pop(job_id, None)
             if future is not None and not future.cancel():
-                LOGGER.warning("Killing a timed out process")
-                future.set_exception(TimeoutError("Task took too long"))
-                pid = self.running_process.pop(task_id, None)
+                LOGGER.warning(f"Killing a job {job_id} with reason: {reason.name}")
```

Comment:
Side note, but it's better to not use f string for logging, but %s arguments, since computation would be skipped depending on level

Reply:
surprisingly Sonar doesn't warn about it...